sched/semaphore/sem_post.c

```diff
@@ -78,21 +78,20 @@ int nxsem_post(FAR sem_t *sem)
   uint8_t proto;
 #endif
 
-  DEBUGASSERT(sem != NULL);
-
   /* The following operations must be performed with interrupts
    * disabled because sem_post() may be called from an interrupt
    * handler.
    */
 
   flags = enter_critical_section();
 
-  sem_count = sem->semcount;
+  sem_count = atomic_fetch_add(NXSEM_COUNT(sem), 1);
 
   /* Check the maximum allowable value */
 
   if (sem_count >= SEM_VALUE_MAX)
     {
```

Comment:
the overflown value is already visible to other threads at this point, isn't it?
isn't it a problem?

Reply:
i guess it's safer to use compare xchg.